db/view/view.cc

```diff
@@ -1293,10 +1281,21 @@ future<> mutate_MV(
             stats.view_updates_pushed_remote += updates_pushed_remote;
             cf_stats.total_view_updates_pushed_remote += updates_pushed_remote;
             future<> view_update = apply_to_remote_endpoints(*target_endpoint, std::move(remote_endpoints), mut, base_token, view_token, allow_hints, tr_state).then_wrapped(
-                    [target_endpoint,
-                     updates_pushed_remote,
-                     maybe_account_failure = std::move(maybe_account_failure)] (future<>&& f) mutable {
-                return maybe_account_failure(std::move(f), std::move(*target_endpoint), false, updates_pushed_remote);
+                    [s = mut.s, &stats, &cf_stats, tr_state, base_token, view_token, target_endpoint, updates_pushed_remote,
+                            units = sem_units.split(sem_units.count())] (future<>&& f) mutable {
```

Comment:
won't that throw "Cannot take more units than those protected by the semaphore" if both local and remote updates are needed? (like was the case triggering the use-after-move)

Reply:
Otherwise looks good to me.

Reply:
No - the first split that happens would leave 0 units in sem_units, making the second split a no-op. Technically using std::move would do the same, because a moved-from instance of semaphore_units is defined as having 0 units, but I consider use-after-move an antipattern, so I coded it explicitly instead.

Reply:
Right. Missed the fact fact that calling `sem_units.count()` the second time will indeed return zero.

LGTM